File: admin/lib/types/category.ts
export interface Category {
  id: string;
  title: string;
  description: string;
  image_url: string | null;
  created_at: string;
}

export interface CategoryCreate {
  title: string;
  description: string;
}

export interface CategoryUpdate {
  title?: string;
  description?: string;
}

export interface CreateCategoryVars {
  data: CategoryCreate;
  image?: File;
}

export interface EditCategoryVars {
  id: string;
  data: CategoryUpdate;
  image?: File;
}

File: backend/app/models/category.py
from datetime import datetime
from typing import Optional
from beanie import Document
from pydantic import BaseModel, Field

  
class Category(Document):
    title: str
    description: str
    image_url: Optional[str] = None
    created_at: datetime = Field(default_factory=datetime.utcnow)
    
    class Settings:
        name = "categories"


class CategoryCreate(BaseModel):
    title: str
    description: str
    image_url: Optional[str] = None


class CategoryUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    image_url: Optional[str] = None